File: typecast-java/src/main/java/io/typecast/exceptions/BadRequestException.java
package io.typecast.exceptions;

/**
 * Exception thrown when the API returns a 400 Bad Request error.
 * 
 * <p>This typically indicates invalid request parameters.</p>
 */
public class BadRequestException extends TypecastException {
    
    /**
     * Creates a new BadRequestException.
     * 
     * @param message      the error message
     * @param responseBody the raw response body
     */
    public BadRequestException(String message, String responseBody) {
        super(message, 400, responseBody);
    }

    /**
     * Creates a new BadRequestException with a cause.
     * 
     * @param message      the error message
     * @param responseBody the raw response body
     * @param cause        the underlying cause
     */
    public BadRequestException(String message, String responseBody, Throwable cause) {
        super(message, 400, responseBody, cause);
    }
}

File: typecast-php/src/Exceptions/BadRequestException.php
<?php

declare(strict_types=1);

namespace Neosapience\Typecast\Exceptions;

class BadRequestException extends TypecastException
{
    public function __construct(string $message = 'Bad request', ?\Throwable $previous = null)
    {
        parent::__construct($message, 400, $previous);
    }
}
